cpp/src/arrow/compute/api_scalar.cc

```diff
@@ -864,6 +864,9 @@ SCALAR_EAGER_UNARY(Millisecond, "millisecond")
 SCALAR_EAGER_UNARY(Minute, "minute")
 SCALAR_EAGER_UNARY(Month, "month")
 SCALAR_EAGER_UNARY(Nanosecond, "nanosecond")
+SCALAR_EAGER_UNARY(PGDay, "pg_day")
+SCALAR_EAGER_UNARY(PGMonth, "pg_month")
+SCALAR_EAGER_UNARY(PGYear, "pg_year")
```

Comment:
`pg_week` can also be supported to provide the same capabilities as `ISOWeek` and `USWeek`.